src/main/java/spleetwaise/address/model/AddressBook.java

```diff
@@ -76,6 +78,17 @@ public Optional<Person> getPersonByPhone(Phone phone) {
         return Optional.of(filteredPersonList.get(0));
     }
 
+    public Optional<Person> getPersonByIndex(Index index) {
```

Comment:
I think this method could be clearer in stating that the index is based on the filtered person list. A future dev might mistake it for the index of the person list instead. Perhaps `getPersonByFilteredPersonListIndex` would be a good fit.